src/clr-angular/utils/drag-and-drop/droppable/droppable.ts

```diff
@@ -0,0 +1,205 @@
+        // Check draggable and droppable have a matching group key.
+        this.isDraggableMatch = this.checkGroupMatch(dragStartEvent.group);
+
+        // Subscribe to dragMoved and dragEnded only if draggable and droppable have a matching group key.
+        if (this.isDraggableMatch) {
+            this.dragStartEmitter.emit(dragStartEvent);
+            this.dragMoveSubscription = this.eventBus.dragMoved.subscribe((dragMoveEvent: ClrDragEvent<T>) => {
+                this.onDragMove(dragMoveEvent);
+            });
+            this.dragEndSubscription = this.eventBus.dragEnded.subscribe((dragEndEvent: ClrDragEvent<T>) => {
+                this.onDragEnd(dragEndEvent);
+            });
+        }
+    }
+
+    private onDragMove(dragMoveEvent: ClrDragEvent<T>): void {
+        if (!this._isDraggableOver && this.isInDropArea(dragMoveEvent.dropPointPosition)) {
+            this.isDraggableOver = true;
+            const dragEnterEvent = this.transformAndBroadcastEvent(dragMoveEvent, ClrDragEventType.DRAG_ENTER);
+            this.dragEnterEmitter.emit(dragEnterEvent);
+        } else if (this._isDraggableOver && !this.isInDropArea(dragMoveEvent.dropPointPosition)) {
+            this.isDraggableOver = false;
+            const dragLeaveEvent = this.transformAndBroadcastEvent(dragMoveEvent, ClrDragEventType.DRAG_LEAVE);
+            this.dragLeaveEmitter.emit(dragLeaveEvent);
+        }
+
+        this.dragMoveEmitter.emit(dragMoveEvent);
+    }
+
+    private onDragEnd(dragEndEvent: ClrDragEvent<T>): void {
+        if (this._isDraggableOver) {
+            if (dragEndEvent.ghostElement) {
+                // By this point draggable ghost component would have been destroyed,
+                // but the element would be active until its animation completes.
+                // As such, once the ghost is dropped over, we will give it "dropped" class.
+                this.renderer.addClass(dragEndEvent.ghostElement, "dropped");
+            }
+            const dropEvent = this.transformAndBroadcastEvent(dragEndEvent, ClrDragEventType.DROP);
+            this.dropEmitter.emit(dropEvent);
+            this.isDraggableOver = false;
+        }
+        this.dragEndEmitter.emit(dragEndEvent);
+        this.unsubscribeFrom(this.dragMoveSubscription);
+        this.unsubscribeFrom(this.dragEndSubscription);
+        this.isDraggableMatch = false;
+        delete this.clientRect;
+    }
+
+    private transformAndBroadcastEvent(event: ClrDragEvent<T>, newEventType: ClrDragEventType): ClrDragEvent<T> {
+        event.type = newEventType;
+        this.eventBus.broadcast(event);
+        return event;
+    }
+
+    ngOnInit() {
+        this.dragStartSubscription = this.eventBus.dragStarted.subscribe((dragStartEvent: ClrDragEvent<T>) => {
+            this.onDragStart(dragStartEvent);
+        });
+    }
+
+    ngOnDestroy() {
+        this.unsubscribeFrom(this.dragStartSubscription);
```

Comment:
We have this kind of individual subscriptions in many of our components, especially the components/directives that need only one subscription, etc. But this one has multiple subscriptions and they need to be unsubscribed at specific locations, so that's why I have this unsubscribe method just to make the unsubscription more concise.